scripts/flattener.py

```diff
@@ -688,85 +681,40 @@ def reconcile_genes(mfinal_obj, cxg_adata_lst, mfinal_adata_genes):
 	cxg_adata_raw_ensembl.var.index.name = None
+	cxg_adata_raw_ensembl.var_names_make_unique(join = '.')
 
-	# Go through possible collapse ensembl genes, and see which one need to actually be collapsed
-	# Chain redundant genes need to be tracked for redundancy removal
-	for gene in genes_to_collapse_dict.keys():
-		total_genes.append(gene)
-		genes_to_collapse_final[gene] = []
-		chain_redundant_bool = False
-		for symbol in genes_to_collapse_dict[gene].keys():
-			total_genes.append(genes_to_collapse_dict[gene][symbol])
-			if genes_to_collapse_dict[gene][symbol] not in genes_to_collapse_final[gene]:
-				if genes_to_collapse_dict[gene][symbol] not in gene_multi_ensembl:
-					if genes_to_collapse_dict[gene][symbol] in mfinal_adata_genes:
-						genes_to_collapse_final[gene].append(genes_to_collapse_dict[gene][symbol])
-					else:
-						stats['not_collapsed_because_not_in_X'].append(genes_to_collapse_dict[gene][symbol])
-				else:
-					stats['not_collapsed_because_ambiguous'].append(genes_to_collapse_dict[gene][symbol])
-					chain_redundant_bool = True
-			else:
-				stats['already_being_collapsed'].append(genes_to_collapse_dict[gene][symbol])
-		# If gene not found in normalized layer, still need to make sure the gene symbol in raw and normalized are consistent
-		# If only 1 gene symbol in the end remains, no longer need to collapse
-		if len(genes_to_collapse_final[gene]) < 2:
-			if len(genes_to_collapse_final[gene]) == 1:
-				if genes_to_collapse_final[gene][0] not in cxg_adata_raw_ensembl.var.index.to_list():
-					for s in genes_to_collapse_dict[gene]:
-						if genes_to_collapse_dict[gene][s] in cxg_adata_raw_ensembl.var.index.to_list():
-							collapsed_need_to_only_switch_id[genes_to_collapse_dict[gene][s]] = genes_to_collapse_final[gene][0]
-							stats['collapsed_need_to_only_switch_id'].append(genes_to_collapse_dict[gene][s])
-			genes_to_collapse_final.pop(gene, None)
-		if chain_redundant_bool == True:
-			for symbol in genes_to_collapse_dict[gene].keys():
-				if genes_to_collapse_dict[gene][symbol] not in gene_multi_ensembl:
-					chain_redundant_genes.append(genes_to_collapse_dict[gene][symbol])
-
-	cxg_adata_raw_ensembl.var.rename(index=collapsed_need_to_only_switch_id, inplace=True)
-	stats['chain_redundant_genes'].extend(chain_redundant_genes)
-	redundant.extend(chain_redundant_genes)
-
-	stats['gene_multi_ensembl'] = gene_multi_ensembl
-	stats['collapsed'] = list(genes_to_collapse_final.keys())
-	stats['redundant'] = list(set(redundant))
 	for key in stats:
 		stats[key] = set(stats[key])
 		overlap_norm = set(mfinal_adata_genes).intersection(stats[key])
 		logging.info("{}\t{}\t{}\t{}\t{}".format(key, len(stats[key]), len(overlap_norm), overlap_norm, stats[key]))
 
-	return cxg_adata_raw_ensembl, genes_to_collapse_final, redundant, genes_to_collapse_dict
+	all_remove = list(set(redundant + multiple_ensembl + multiple_symbols))
+
+	return cxg_adata_raw_ensembl, redundant, all_remove
```

Comment:
if going with the above suggestion, would need to update this to `stats['redundant']`